plugins/kubernetes/test/models/kubernetes/build_job_executor_test.rb

```diff
@@ -0,0 +1,152 @@
+            namespace: 'test_ns',
+            labels: labels
+          },
+          spec: {
+            template: {
+              metadata: {
+                name: 'test_job',
+                namespace: 'test_ns',
+                labels: labels
+              },
+              spec: {
+                containers: [{ name: 'test-job' }]
+              }
+            }
+          }
+        })
+      end
+
+      before do
+        extension_client.expects(:delete_job).once
+        Kubernetes::RoleConfigFile.expects(:new).returns build_config
+        # build_config.job.stubs(:first_container).returns job_container_spec
+      end
+
+      around { |t| with_env "KUBE_BUILD_JOB_FILE": build_job_config_path, &t }
+
+      it 'reads job config file and populate correct job parameters, but fails to start a job' do
+        extension_client.expects(:create_job).raises(KubeException.new(500, 'Server Error', {}))
+
+        assert_raises KubeException do
+          execute!(registry_info: registry_info)
+        end
+
+        project_name_dash = project.permalink.tr('_', '-')
+        assert_match(/^#{Regexp.quote(project_name_dash)}-docker-build-#{build.id}-[0-9a-f]{14}$/,
+          build_config.job[:metadata][:name])
+
+        labels_config = [build_config.job[:metadata][:labels], build_config.job[:spec][:template][:metadata][:labels]]
+        labels_config.each do |v|
+          assert_equal v[:project], project_name_dash
+          assert_equal v[:role], 'docker-build-job'
+          assert_equal v[:foo], 'bar'
+        end
+
+        assert_equal build_config.job[:spec][:template][:spec][:containers][0][:args],
+          [project.repository_url, build.git_sha, project.docker_repo, 'latest', 'no', 'no']
+        assert_equal build_config.job[:spec][:template][:spec][:containers][0][:env].length, 1
+      end
+
+      describe 'can create job' do
+        let(:job_api_obj) { stub }
+        let(:job_pod) { stub(first: stub(name: stub, namespace: stub)) }
+        let(:job_pod_log) { ['A long long log', 'A shorter log'] }
+        let(:pod_api_obj) { stub(name: stub, namespace: stub) }
+
+        before do
+          extension_client.expects(:create_job).once
+          extension_client.expects(:get_job).at_least_once
+          client.expects(:get_pods).returns(job_pod).at_least_once
+          client.expects(:watch_pod_log).with(
+            pod_api_obj.name, pod_api_obj.namespace
+          ).returns(job_pod_log).at_least_once
+          Kubernetes::Api::Job.expects(:new).returns(job_api_obj).at_least_once
+          Kubernetes::Api::Pod.expects(:new).returns(pod_api_obj).at_least_once
+          job_api_obj.stubs(:name).returns 'job-123'
+        end
+
+        it 'job is unsuccessful' do
+          job_api_obj.stubs(:failure?).returns true
+
+          success, job_log = execute!(registry_info: registry_info)
+          refute success
+          assert_empty job_log
+        end
+
+        it 'job gets timeout' do
```

Comment:
strange test case name ... how about something like `it does xyz when the job times out`